### PR TITLE
Sprite Movement working with AI movement

### LINKED ISSUE #32187: When mobs move themselves, there is no movement animation
## Description
When you control the mob, moving animation provided by SpriteMovementComponent is working perfectly. When the mob is controlled by the game, movement animation not shown. 

**Reproduction**
See an attached video

**Screenshots**
https://github.com/user-attachments/assets/313823f4-4377-4a04-9d00-73faf485d189
**Additional context**
Seems like the problem happens because in SpriteMovementSystem the game subscribes only for user's input and not for mob movement event. Do we have one, btw?
`
        SubscribeLocalEvent<SpriteMovementComponent, MoveInputEvent>(OnSpriteMoveInput);
`

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR

Now AI movement can work with SpriteMovementSystem

## Why / Balance

bug 
fix #32187 

## Media

https://github.com/user-attachments/assets/630c46b4-2546-4ea3-96d1-8b50cfbf074d

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

**Changelog**
to minor

### CODE REVIEW

**Comment by IProduceWidgets** [comment]
Mostly commenting to follow so I can implement once merged. 

The code is surprisingly straight forward though

File: Content.Shared/Movement/Systems/SharedSpriteMovementSystem.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        SubscribeLocalEvent<SpriteMovementComponent, SpriteMoveEvent>(OnSpriteMoveInput);
+    }
+
+    private void OnSpriteMoveInput(Entity<SpriteMovementComponent> ent, ref SpriteMoveEvent args)
+    {
+        if (ent.Comp.IsMoving == args.IsMoving)
+            return;
+
+        ent.Comp.IsMoving = args.IsMoving;
+        Dirty(ent);
```
Comment (metalgearsloth) [inline Content.Shared/Movement/Systems/SharedSpriteMovementSystem.cs:21]: I don't think this even needs to be networked? It's just getting derived from inputmover data.

Comment (TheShuEd) [reply Content.Shared/Movement/Systems/SharedSpriteMovementSystem.cs:21]: SteeringSystem serveronly

Comment (TheShuEd) [reply Content.Shared/Movement/Systems/SharedSpriteMovementSystem.cs:21]: Oh, i get it, yes, not needed for input mover

Comment (TheShuEd) [reply Content.Shared/Movement/Systems/SharedSpriteMovementSystem.cs:21]: @metalgearsloth  If I remove Dirty(), it won't work anymore. Neither when the creature is controlled by the AI, nor when the player

**Review by SlamBamActionman — APPROVED** [review]
I've tested this ingame with various other creatures. The change to spider parentage is possibly a bit out of scope for the PR, but still works.

**Pull request merged**